core/src/main/java/org/geysermc/geyser/translator/protocol/java/entity/player/JavaPlayerPositionTranslator.java

```diff
@@ -150,11 +150,14 @@ public void translate(GeyserSession session, ClientboundPlayerPositionPacket pac
     }
 
     private void acceptTeleport(GeyserSession session, double x, double y, double z, float yaw, float pitch, int id) {
-        // Confirm the teleport when we receive it to match Java edition
-        ServerboundAcceptTeleportationPacket teleportConfirmPacket = new ServerboundAcceptTeleportationPacket(id);
-        session.sendDownstreamPacket(teleportConfirmPacket);
-        // Servers (especially ones like Hypixel) expect exact coordinates given back to them.
-        ServerboundMovePlayerPosRotPacket positionPacket = new ServerboundMovePlayerPosRotPacket(false, x, y, z, yaw, pitch);
-        session.sendDownstreamPacket(positionPacket);
+        // Teleports with id 0 are never used in Vanilla, but ViaVersion uses it when the server doesn't support teleport ids yet.
+        if (id != 0) {
```

Comment:
I'm not sure how I feel about this, because we are breaking what vanilla does in this scenario. And unfortunately I don't think we can use the tag hacks we use for other versions. But you also said this happens on Java 1.19.2 or no?

Reply:
I'll revert it, because if a player gets teleported more than Integer.MAX_VALUE, this will cause a problem